power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_se_solver.hpp

```diff
@@ -348,38 +315,170 @@ template <bool sym> class NewtonRaphsonSESolver {
+        rhs_block.tau_q() -= imag(f_x_complex_row);
+    }
+
+    void process_shunt_measurement(NRSEGainBlock<sym>& block, NRSERhs<sym>& rhs_block, auto const& yii,
+                                   auto const& u_state, auto const& measured_power) {
+        auto const hm_ui_ui_yii = hm_complex_form(yii, u_state.ui_ui_conj);
+        auto const nl_ui_ui_yii = dot(hm_ui_ui_yii, u_state.abs_ui_inv);
+        auto const f_x_complex = sum_row(hm_ui_ui_yii);
+        auto const f_x_complex_abs_ui_inv = sum_row(nl_ui_ui_yii);
+
+        auto jac_block = calculate_jacobian(hm_ui_ui_yii, nl_ui_ui_yii);
+        jac_block += jacobian_diagonal_component(f_x_complex_abs_ui_inv, f_x_complex);
+        auto const& block_F_T_k_w = transpose_multiply_weight(jac_block, measured_power);
+        multiply_add_jacobian_blocks_lhs(block, block_F_T_k_w, jac_block);
+        multiply_add_jacobian_blocks_rhs(rhs_block, block_F_T_k_w, measured_power, f_x_complex);
+    }
+
+    /**
+     * @brief Adds contribution of branch measurements to the G_(r, r), G_(r, c) and eta_r blocks,
+     *  given the iteration passes through (r, c) ie. row, col
+     *
+     * When iterating via (row, col), have 4 cases regarding branch measurements:
+     *      if y_type == yft
+     *          if from_measurement,
+     *              xi = f, mu = t, chi = row, psi = col
+     *          if to_measurement,
+     *              xi = t, mu = f, chi = col, psi = row
+     *      if y_type == ytf &
+     *          if from_measurement,
+     *              xi = f, mu = t, chi = col, psi = row
+     *          if to_measurement,
+     *              xi = t, mu = f, chi = row, psi = col
+     *
+     * f_P_(x) = -M(U_chi, U_chi, Y_xi_xi) - M(U_chi, U_psi, Y_xi_mu)
+     * f_Q_(x) = H(U_chi, U_chi, Y_xi_xi) + H(U_chi, U_psi, Y_xi_mu)
+     *
+     *
+     * @param block G_(r, c)
+     * @param diag_block G_(r, r)
+     * @param rhs_block RHS(r)
+     * @param y_xi_xi
+     * @param y_xi_mu
+     * @param u_state voltage state vector
+     * @param order bool to determine if (chi, psi) = (row, col) or (col, row)
+     * @param measured_power
+     */
+    void process_branch_measurement(NRSEGainBlock<sym>& block, NRSEGainBlock<sym>& diag_block, NRSERhs<sym>& rhs_block,
+                                    const auto& y_xi_xi, const auto& y_xi_mu, const auto& u_state, auto const order,
+                                    const auto& measured_power) {
+        auto const hm_u_chi_u_chi_y_xi_xi = hm_complex_form(y_xi_xi, u_state.u_chi_u_chi_conj(order));
+        auto const nl_u_chi_u_chi_y_xi_xi = dot(hm_u_chi_u_chi_y_xi_xi, u_state.abs_u_chi_inv(order));
+
+        auto const hm_u_chi_u_psi_y_xi_mu = hm_complex_form(y_xi_mu, u_state.u_chi_u_psi_conj(order));
+        auto const nl_u_chi_u_psi_y_xi_mu = dot(hm_u_chi_u_psi_y_xi_mu, u_state.abs_u_psi_inv(order));
+
+        auto const f_x_complex = sum_row(hm_u_chi_u_chi_y_xi_xi + hm_u_chi_u_psi_y_xi_mu);
```

Comment:
I think this is wrong. Shouldn't this be:
```suggestion
        auto const f_x_complex = sum_row(hm_u_chi_u_chi_y_xi_xi + hm_u_chi_u_chi_y_xi_mu + hm_u_chi_u_psi_y_xi_xi + hm_u_chi_u_psi_y_xi_mu);
```
See: 
![image](https://github.com/PowerGridModel/power-grid-model/assets/66305765/b4e696b7-2da1-4301-b8fe-02261f1c7d8b)
You sum over `i,j` and over `f,t` separately, so there should be 4 possibilities: `i+f`, `i+t`, `j+f`, `j+t`

Reply:
The f(x) is the power from the from side ie. the one which flows from `xi`. Hence two components. Maybe there is a better way of writing this. Thinking.